site/zenodo_rdm/legacy/serializers/schemas/legacyjson.py

```diff
@@ -136,14 +136,12 @@ def dump_owner(self, obj):
     def dump_files(self, obj):
         """Dump files."""
         result = []
-        bucket_url = obj["links"].get("bucket")
         files_url = obj["links"].get("files")
         for key, f in obj["files"].get("entries", {}).items():
-            file_id = f["id"]
             if files_url:
-                links = {"self": f"{files_url}/{file_id}"}
-            if bucket_url:
-                links["download"] = f"{bucket_url}/{key}"
+                self_html = obj["links"]["self_html"]
+                links = {"self": f"{files_url}/{key}"}
+                links["download"] = f"{self_html}/files/{key}"
```

Comment:
@lnielsen suggested to put here the html file download link as it is more likely to keep consistent.

Reply:
DIscussed with @slint: we cannot do this because then other UIs fetching files from that link will hit CORS issues.